pkg/scaffold/v1/resource/resource.go

```diff
@@ -34,6 +34,10 @@ type Resource struct {
 	// Group is the API Group.  Does not contain the domain.
 	Group string
```

Comment:
Does it make sense if we enforce no `-` in `Group` field?
Then we don't need to introduce a new `GroupImportSafe` field.

Reply:
Hi @mengqiy, 

Note that the correct validation for the group is the func, so users should be able to create groups as `myapp-operator` for example.  See more over it is https://godoc.org/k8s.io/apimachinery/pkg/util/validation#IsDNS1123Subdomain. (done a PR for it as well see https://github.com/kubernetes-sigs/kubebuilder/issues/1040)
 Also, see more about this here: https://github.com/kubernetes-sigs/kubebuilder/issues/1007#issuecomment-534455225